### PR TITLE
Completed

### PR DESCRIPTION
**Pull request by samgaco**

### CODE REVIEW

**Comment by dipto0321** [comment]
HI @samgaco and @Forison ,
Congrats for the first try!. Here is the app behavior should

#### Behaviour
- [x] User can sign in
- [ ] User can sign up
- [ ] As a signed-in user you can create a new post. Please add a menu or link in the view so that user can click it and it redirects him to that path.
- [ ] As a signed-in user you can see post's author. Also, add a link for this path.
- [x] User can sign_out
- [ ] As a guest you can see only post's text. Add this path in the menu

Also, take a look at these points: 
- Where is the association with the user and post in the user model? Please review those active records lessons from rails guide again if needed. Don't need to hurry. 🙂 
- Indentation is not correct. Please use rubocop to fix those. check rubocop documentation to do that.
- [Here ](https://medium.com/podiihq/understanding-rails-routes-and-restful-design-a192d64cbbb5) is a nice article about rails routing. Let's implement **Rails RESTful Design** (Which is mostly recommended one) for users and posts. All those routes for the session are okay. So don't touch those. Want to know more about routing check this [guideline](https://guides.rubyonrails.org/routing.html).
 
It will be better if you add some design for this project but it's completely your choice. Not mandatory. But please add those features which are really important. Try to treat this app like your one of the dream project or a real project.

After completing those task please send another request to final review this project. 

**Happy Coding!**

[**Note**: If you need any clarification for this feedback or have any question then leave a reply here (Don't use quote reply option. Just add reply simply) or knock me on slack]

Thanks 🙏

File: app/models/user.rb [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    self.remember_token = User.new_token
+    update_attribute(:remember_digest, User.digest(remember_token))
+  end
+
+  def create_activation_digest
+    self.activation_token = User.new_token
+    self.remember_digest = User.digest(activation_token)
+  end
+
+  def self.digest(string)
```
Comment (adriaanbd) [inline app/models/user.rb]: Project specifications indicate the use of `Digest::SHA1.hexdigest` and not `BCrypt`. Please update this.

File: config/routes.rb [hunk 2]
```diff
@@ -1,3 +1,15 @@
+# frozen_string_literal: true
+
 Rails.application.routes.draw do
+  # get 'users/new'
+  resource :users
+  get 'posts/new'
```
Comment (adriaanbd) [inline config/routes.rb]: Here you could use:

[Required]
```
resources :posts, only: [:new, :create, :index]
```

Try to use this way of writing routes whenever possible.

**Comment by adriaanbd** [comment]
Hey! After discussing with the TSE team about the obligatory use Digest::SHA1, we have decided to modify the requirements to allow either of these two ways of accomplishing the same task. Thus, please disregard the [Required] tag for the use of `Digest::SHA1`.

**Pull request merged**